crates/autopilot/src/domain/settlement/solution/mod.rs

```diff
@@ -194,6 +200,16 @@ pub mod error {
     }
 }
 
+/// Fee per trade in a solution. Contains breakdown of protocol fees and total
+/// fee.
+#[derive(Debug, Clone)]
+pub struct OrderFee {
+    /// Gas fee + protocol fees. Total fee taken for the execution of the order.
+    pub total: eth::SellTokenAmount,
```

Comment:
Is this for backwards compatibility? Otherwise I would find it more straight forward to explicitly list the parts of the fee (protocol + network) and expose a total() method on the struct.

Reply:
protocol fee is in surplus token. It doesn't have to be, but here is the discussion around it: https://github.com/cowprotocol/services/issues/2862#issuecomment-2273548210

Reply:
The frontend wants to display for each trade two things
1. Protocol Fee
2. Network fee

What's the most straight forward way to achieve this if tokens are denominated in different currencies? Do we need to explicitly add the network fee to this struct (and compute it from some prices we have available) or can the protocol fee be converted into sell token?

Reply:
> What's the most straight forward way to achieve this if tokens are denominated in different currencies?

My plan was to save protocol fees in surplus token (but also save surplus token so we can change this if needed) as this is the most correct way as felix h. mentioned.
1. For frontend, we can convert protocol fees to sell token on `get_trades`
2. For solver team, they can fetch order_execution.protocol_fees directly

If we want to ignore 2, then yes, we can do the conversion in `domain::Settlement` immediately and do a proper network / protocol fees separation (although it still won't be nicely like this in database)

Reply:
Ideally, both order_execution::executed_surplus_fee and these protocol fees are expressed in surplus token, but we would need a db migration for this table to convert the executed_surplus_fee to surplus token

Reply:
Converted protocol fees into sell token and refactored into <network, protocol>